addons/xterm-addon-web-links/src/WebLinkProvider.ts

```diff
@@ -3,20 +3,35 @@
  * @license MIT
  */
 
-import { ILinkProvider, IBufferCellPosition, ILink, Terminal } from 'xterm';
+import { ILinkProvider, IBufferCellPosition, ILink, Terminal, ILinkMatcherOptions, IViewportRange } from 'xterm';
 
 export class WebLinkProvider implements ILinkProvider {
 
   constructor(
     private readonly _terminal: Terminal,
     private readonly _regex: RegExp,
-    private readonly _handler: (event: MouseEvent, uri: string) => void
+    private readonly _handler: (event: MouseEvent, uri: string) => void,
+    private readonly _options: ILinkMatcherOptions
   ) {
 
   }
 
   public provideLinks(y: number, callback: (links: ILink[] | undefined) => void): void {
-    callback(LinkComputer.computeLink(y, this._regex, this._terminal, this._handler));
+    const links = LinkComputer.computeLink(y, this._regex, this._terminal, this._handler);
+    callback(this._addCallBacks(links, this._options.tooltipCallback, this._options.leaveCallback));
+  }
+
+  private _addCallBacks(links: ILink[], hover?: (event: MouseEvent, uri: string, location: IViewportRange) => void, leave?: (event: MouseEvent, uri: string) => void): ILink[] {
+    return links.map((link): ILink => {
```

Comment:
I think we can avoid the type annotation?

```suggestion
    return links.map(link => {
```